x-pack/plugins/observability_solution/observability_ai_assistant/server/service/client/operators/continue_conversation.ts

```diff
@@ -114,27 +114,31 @@ function executeFunctionAndCatchError({
 function getFunctionDefinitions({
   functionClient,
   functionLimitExceeded,
+  disableFunctions,
 }: {
   functionClient: ChatFunctionClient;
   functionLimitExceeded: boolean;
+  disableFunctions: boolean;
 }) {
-  const systemFunctions = functionLimitExceeded
-    ? []
-    : functionClient
-        .getFunctions()
-        .map((fn) => fn.definition)
-        .filter(
-          (def) =>
-            !def.visibility ||
-            [FunctionVisibility.AssistantOnly, FunctionVisibility.All].includes(def.visibility)
-        );
+  const systemFunctions = functionClient
+    .getFunctions()
+    .map((fn) => fn.definition)
+    .filter(
+      (def) =>
+        !def.visibility ||
+        [FunctionVisibility.AssistantOnly, FunctionVisibility.All].includes(def.visibility)
+    );
 
   const actions = functionLimitExceeded ? [] : functionClient.getActions();
 
   const allDefinitions = systemFunctions
     .concat(actions)
     .map((definition) => pick(definition, 'name', 'description', 'parameters'));
 
+  if (disableFunctions || functionLimitExceeded) {
+    return allDefinitions.filter((def) => def.name === 'context');
+  }
```

Comment:
We should keep the context function because we inject the request ourselves, so need to be able to call it.

Reply:
I don't think this is necessary though, because these are the functions that we pass to the LLM. We also have some places where we check if there are any functions available and if not, change the instructions for the LLM. So I think we're better off leaving them out entirely here as well. (It does make me realize it's a little hard to understand how we inject `context` function calls now).

Reply:
If I remove the context function from here I get this error:

```
Error: Function context called but was not available
    at createFunctionNotFoundError (conversation_complete.ts:174:10)
    at Object.next (fail_on_non_existing_function_call.ts:33:46)
```

It felt cleaner to leave the context function in rather than removing it, and then handle this exception. WDYT?